numpy/core/src/common/simd/avx512/conversion.h

```diff
@@ -25,6 +25,10 @@
 #endif
 #define npyv_cvt_s32_b32 npyv_cvt_u32_b32
 #define npyv_cvt_s64_b64 npyv_cvt_u64_b64
+
+#define npyv_cvt_f16_f32 _mm512_cvtph_ps 
+#define npyv_cvt_f32_f16 _mm512_cvtps_ph 
+
```

Comment:
```suggestion
```

Since the implementation is limited to AVX512, so it is better not to add these intrinsics to the branch `simd/avx512` and move them to [loops_umath_fp.dispatch.c.src](https://github.com/numpy/numpy/pull/21955/files#diff-a6fc1c1638228797b18fa34b279d9009e036c3748a7c2abc489c48d08fce1a0b) instead.

Reply:
I would like to leave them there. [SSE and AVX ](https://www.intel.com/content/www/us/en/docs/intrinsics-guide/index.html#text=fp16c&ig_expand=2191) also have these _cvt_ instructions and these can be added to AVX/SSE if and when required or used.

Reply:
To keep the workflow organized, adding new universal intrinsics should cover other architectures, also necessary to provide a test case for them too within `core/tests/test_simd.py`. supporting `SSE`, `VSX2`, `ARMHF/NEON` will require emulated intrinsics for half <-> single precision conversion, a lot of work that is better implemented in a separate pull request.

Reply:
> To keep the workflow organized, adding new universal intrinsics should cover other architectures, also necessary to provide a test case for them too within `core/tests/test_simd.py`. supporting `SSE`, `VSX2`, `ARMHF/NEON` will require emulated intrinsics for half <-> single precision conversion, a lot of work that is better implemented in a separate pull request.

Fine with me. Moved the intrisincs to the local dispatch file.